public/app/features/canvas/elements/server.tsx

```diff
@@ -0,0 +1,474 @@
+import { css } from '@emotion/css';
+import React from 'react';
+
+import { GrafanaTheme2 } from '@grafana/data';
+import { useStyles2 } from '@grafana/ui';
+import { ColorDimensionConfig, DimensionContext, ScalarDimensionConfig } from 'app/features/dimensions';
+import { ColorDimensionEditor, ScalarDimensionEditor } from 'app/features/dimensions/editors';
+
+import { CanvasElementItem, CanvasElementProps } from '../element';
+
+interface ServerConfig {
+  blinkRate?: ScalarDimensionConfig;
+  statusColor?: ColorDimensionConfig;
+  bulbColor?: ColorDimensionConfig;
+  type: ServerTypes;
+}
+
+interface ServerData {
+  blinkRate?: number;
+  statusColor?: string;
+  bulbColor?: string;
+  type: ServerTypes;
+}
+
+enum ServerTypes {
+  Single = 'Single',
+  Stack = 'Stack',
+  Database = 'Database',
+  Terminal = 'Terminal',
+}
+
+type Props = CanvasElementProps<ServerConfig, ServerData>;
+
+const ServerDisplay = ({ data }: Props) => {
+  const styles = useStyles2(getStyles);
+
+  const bulbColor = data?.bulbColor;
+  const bulbAnimation = `blink ${data?.blinkRate ? 1 / data.blinkRate : 0}s infinite step-end`;
+
+  return (
+    <svg viewBox="0 0 207.95 197.78">
+      {data?.type === ServerTypes.Stack ? (
+        <>
```

Comment:
This could be cleaned up a bit - potentially we could create a separate folder for the server element with a nested directory where we could export each of these svg fragments from their own file? Or maybe there is another approach we can take to reduce repeated svg fragments

Reply:
Some can be consolidated into general use, but most separated; but that may depend on if we decide to change up the background styling.

Reply:
Refactored to separate each type.